api/src/main/java/ai/djl/modality/cv/translator/YoloV5Translator.java

```diff
@@ -0,0 +1,271 @@
+               IntermediateResult detection = detections[j];
+               final Rectangle2D location = detection.getLocation();
+               if (boxIou(rec, location) < nmsThresh) {
+                  pq.add(detection);
+               }
+            }
+         }
+      }
+      return new DetectedObjects(retClasses, retProbs, retBB);
+   }
+
+   protected double overlap(double x1, double w1, double x2, double w2) {
+      double l1 = x1 - w1 / 2;
+      double l2 = x2 - w2 / 2;
+      double left = Math.max(l1, l2);
+      double r1 = x1 + w1 / 2;
+      double r2 = x2 + w2 / 2;
+      double right = Math.min(r1, r2);
+      return right - left;
+   }
+
+   private DetectedObjects processFromBoxOutput(TranslatorContext ctx, NDList list) {
+      final NDArray ndArray = list.get(0);
+      final ArrayList<IntermediateResult> intermediateResults = new ArrayList<>();
+      for (long i = 0; i < ndArray.size(0); i++) {
+         final float[] boxes = ndArray.get(i).toFloatArray();
+         float maxClass = 0;
+         int maxIndex = 0;
+         final float[] clazzes = new float[classes.size()];
+         System.arraycopy(boxes, 5, clazzes, 0, clazzes.length);
+         for (int c = 0; c < clazzes.length; c++) {
+            if (clazzes[c] > maxClass) {
+               maxClass = clazzes[c];
+               maxIndex = c;
+            }
+         }
+         final float score = maxClass * boxes[4];
+         if (score > threshold) {
+            final float xPos = boxes[0];
+            final float yPos = boxes[1];
+            final float w = boxes[2];
+            final float h = boxes[3];
+            intermediateResults.add(new IntermediateResult(classes.get(maxIndex), score, maxIndex, new Double(Math.max(0, xPos - w / 2), Math.max(0, yPos - h / 2), w, h)));
+         }
+      }
+      return nms(intermediateResults);
+   }
+
+   public DetectedObjects processFromDetectOutput(TranslatorContext ctx, NDList list) {
+      throw new RuntimeException("detect layer output is not supported yet, check correct YoloV5 export format");
+   }
+
+   /**
+    {@inheritDoc}
+    */
+   @Override
+   public DetectedObjects processOutput(TranslatorContext ctx, NDList list) {
+      switch (yoloOutputLayerType) {
+         case BOX:
+            return processFromBoxOutput(ctx, list);
+         case DETECT:
+            return processFromDetectOutput(ctx, list);
+         case AUTO:
+            if (list.get(0).getShape().dimension() > 2) {
+               return processFromDetectOutput(ctx, list);
+            }
+      }
+      return processFromBoxOutput(ctx, list);
+   }
+
+   /**
+    The builder for {@link YoloV5Translator}.
+    */
+   public static class Builder extends ObjectDetectionBuilder<YoloV5Translator.Builder> {
+      public Builder() {
+         // custom pipeline to match default YoloV5 input layer
+         pipeline = new Pipeline().add(array -> array.transpose(2, 0, 1).toType(DataType.FLOAT32, false).div(255));
```

Comment:
Move this to build() method.
pipeline could be manually set in the builder, We should create pipeline only if user didn't set pipeline.